README.md

```diff
@@ -127,10 +127,11 @@ preview.show(fileId, accessToken, {
 | collection |  | List of file IDs to iterate over for previewing |
 | header | 'light' | String value of 'none' or 'dark' or 'light' that controls header visibility and theme |
 | logoUrl |  | URL of logo to show in header |
-| showAnnotations | false | Whether annotations and annotation controls are shown. This option will be overridden by viewer-specific annotation options if they are set. See [Box Annotations](https://github.com/box/box-annotations) for more details. |
+| showAnnotations | false | Whether annotations and annotation controls are shown. This option will be overridden by viewer-specific annotation options if they are set. See [Box Annotations](https://github.com/box/box-annotations) for more details |
 | showDownload | false | Whether download button is shown |
 | useHotkeys | true | Whether hotkeys (keyboard shortcuts) are enabled |
 | pauseRequireJS | false | Temporarily disables requireJS to allow Preview's third party dependencies to load |
+| fileVersions | {} | Mapping of file ID to file version ID to preview. Each file ID key should map to a valid non-current file version ID for that file. If no mapping is passed, or the file being previewed has no key/value pair in this map, we assume the current file version |
```

Comment:
I am wondering if the API here is getting confusing and complicated. Can we just re-use the `collection` attribute and document that you can pass in a `file` object and make them work with versions too?

So if you want to show a specific version, you do

const file = { id: 123, file_version: { id: 123 } }; // well formed (but incomplete)
preview.show(file, token, { ... })

Likewise collection already takes fully formed file objects.

Reply:
I think the use case of an incomplete file object (or even a fully formed file object) is not going to be very common for external developers. IMO:

```
preview.show(FILE_ID, ACCESS_TOKEN, {
    fileVersions: {
        FILE_ID: FILE_VERSION_ID
    }
});
```

is easier to understand than:

```
preview.show({ id: FILE_ID, file_version: { id: FILE_VERSION_ID } }, ACCESS_TOKEN, {
    useFileVersion: true
});
```

I think most people are going to be working with file IDs and file version IDs, and having to jigger them into our internal file format isn't going to be commonplace.

Thoughts @jeremypress @pramodsum @JustinHoldstock @DanDeMicco?

Reply:
@tonyjin there is definitely some overlap in the design of this, and what I am working on

The question is, for every option we support that is specific to what preview renders for a given FILE_ID, do we want to add in a new property that is of this format? My opinion is that will not be ideal.

What if we take what @priyajeet mentioned and have a well defined file type such as
```
id : string (required),
version : number (optional),
foo : string (optional),
bar : string (optional),
```

and we can then use the file object anywhere a FILE_ID string is currently used, while maintaining backwards compatibility.

For example:

The current API :
```javascript
preview.show(FILE_ID_1, ACCESS_TOKEN, {})
```
and
```javascript
preview.show(FILE_ID_1, ACCESS_TOKEN, {
   collection : [FILE_ID_1, FILE_ID_2]
})
```
will continue work

but if they want more options they can use:
```javascript
const file = { id: 123, file_version: 2 }; // well formed (but incomplete)
preview.show(file, ACCESS_TOKEN, {});
```

and if they want to apply it for collections:
```javascript
const file1 = { id: 123, file_version: 2 };
const file2 = { id: 456, file_version: 3, foo: "bar" };
preview.show(file, ACCESS_TOKEN, {
   collection : [file1, file2]
});
```

they can also be mixed and matched
```javascript
const file2 = { id: 456, file_version: 3, foo: "bar" };
preview.show(FILE_ID_1, ACCESS_TOKEN, {
   collection : [FILE_ID_1, file2]
});
```

Reply:
Ah I see what you mean now, let me think about this.

Reply:
I think its incorrect to assume that people will be dealing with just file ids and file version ids.
They will mostly dealing with objects that our API returns (complete or incomplete) since that's where the ids will come in the first place. As such preview interface should be able to take those objects readily along with ids (string id should signify the latest).

We should also not create our own objects as shown above, because we don't want people to learn yet another object structure. They should use existing objects that our returned by the API, so maybe instead of a flag...

User gets a file object either via /file API or more likely /folder API.

```
const file = {
    "type": "file",
    "id": "5000948880",
    "file_version": {
        "type": "file_version",
        "id": "26261748416",
        "sha1": "134b65991ed521fcfe4724b7d814ab8ded5185dc"
    },
    ...
};
preview.show(file, ...) // Uses the file and version to get in this case just the latest.
```

The above shows the file with the specified latest version. But the user may only want to see another older version(s). Since they don't know which version, they use /versions API to get versions which returns the array or versions as shown below. But now they want to preview all prior versions, they would do something like


File Version
```js
const fileVersions = [
        {
            "type": "file_version",
            "id": "46313006161",
            ...
        },
        {
            "type": "file_version",
            "id": "41550732434",
            ...
    ];

const versions = fileVersions.map((version) => Object.assign({}, file, { file_version: version });
preview.show(versions[0], token, {
    collection: versions
})
```